language/tools/move-cli/Cargo.toml

```diff
@@ -12,6 +12,7 @@ edition = "2018"
 [dependencies]
 anyhow = "1.0.31"
 difference = "2.0.0"
+serial_test = "0.5.0"
```

Comment:
I realized that if we run the tests concurrently, we may see an error like
thread 'test_run_one_file' panicked at '`move clean` failed to eliminate move_data directory', language/tools/move-cli/src/test.rs:114:

My guess is that this is caused by the concurrently running the same test creates some conflicts, so I added the `#[serial]` tags in the tests.

This `serial_test` is used in `network/src/protocols` so is not technically a new introduction to the whole project. But please let me know if this is not preferred.

Reply:
I don't understand the issue. All of the tests should have their own move_data directories so I'm confused as to why there is any concurrency issue

Reply:
I think the problem is that I suggested that we reuse dirs tested in `run_all` in `test_run_one_dir` and `test_run_one_file`. I think we should probably either delete these tests (I suggested them but am not wedded to them) or create separate directories for them that are outside of the paths covered by `test_run_all`.

Reply:
(i.e., the race is `run_all` || `test_run_one_dir`)

Reply:
Sam is right, it is actually three way: `run_all` || `test_run_one_dir` || `test_run_one_file` since that particular file also resides in that dir.

Reply:
Ah! I see... but am torn about what to do. 

I'd probably say its best to have separate directories. Otherwise, we basically are rerunning all of the tests 4 separate times on each CI/`cargo test` run, which seems overly paranoid. 

If I had to pick, I'd say either
- Make seprate files/dirs for these 3 functions
- kill all 3: `test_run_all`, `test_run_one_dir`, and `test_run_one_file` and just add tests that call out to the `test` CLI command

Reply:
Per each CI/`cargo test` it should be running all tests in `testsuite` plus 2 extra, the one for `run_one_dir` and the one for `run_one_file`.

Reply:
That still seems crazy. Especially as these tests grow larger... Anything over just one time seems bad

Reply:
Right. So shall I just create a few mock dirs under, say `move-cli/tests/unit-test` to cover the three cases?

Reply:
You could... 

I think it would be fine to just use the existing test system and not have these three functions at all. 

So you would add an `args.txt`  that would have
```
test --path <path to subdir>
```

Reply:
That sounds good! I don't lock `mocktests` but can't think of a better name (I thought of `metatests` because they're testing the testing functionality, but that is maybe confusing)...

Reply:
Question: the approach of adding an `args.txt` with `test --path <path to subdir>` will not be scheduled by `cargo test`? We still need sth in `cli_testsuite.rs` for `cargo` to run the test

Reply:
Correct. I was thinking that the `test_run_one_dir` and `test_run_one_file` #[test]` functions could live in `cli_testsuite.rs` (similar to what you had before). That will let `cargo test` pick them up.

Reply:
I added an `args.txt` in the new `metatests` folder which outlines effectively the three cases
```
test dummy/test_1/args.txt
test dummy/test_2
test dummy
```
And left a pointer in `cli_testsuite.rs` to invoke the `args.txt`.